src/config/machine.rs

```diff
@@ -4,30 +4,100 @@ use derivative::Derivative;
 use serde::{Deserialize, Serialize};
 
 /// Machine configuration.
-// TODO: Provide a builder for `Machine`.
-#[derive(Derivative, Debug, Serialize, Deserialize, Default)]
+#[derive(Derivative, Debug, Serialize, Deserialize)]
 pub struct Machine<'m> {
-    /// Flag for enabling/disabling simultaneous multithreading. Can be enabled only on x86.
-    pub smt: bool,
+    smt: bool,
+    track_dirty_pages: bool,
+    mem_size_mib: i64,
+    vcpu_count: usize,
+    // TODO: Should create a type to validate it like the Go API.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cpu_template: Option<Cow<'m, str>>,
+}
+
+impl<'m> Machine<'m> {
```

Comment:
And `'m` here? I guess why isn't Jailer `'j`?

Reply:
C&P mistakes, Chris. :) I think I first wrote the main `Config` struct and then some others I started with c&p of that or other ones. I can make them consistent sure.